cube/src/main/java/zone/cogni/asquare/cube/convertor/data2shacl/ShaclGenerator.java

```diff
@@ -299,41 +276,31 @@ private Literal getOneAsInteger() {
     return ResourceFactory.createTypedLiteral("1", XSDDatatype.XSDinteger);
   }
 
-  private void setNodeKind(Configuration configuration,
-                           RdfStoreService rdfStoreService,
-                           Model shacl,
-                           Resource targetClass,
-                           Resource path,
-                           Resource propertyShape) {
+  private void setNodeKind(@Nonnull Configuration configuration,
+                           @Nonnull RdfStoreService rdfStore,
+                           @Nonnull Model shacl,
+                           @Nonnull Resource targetClass,
+                           @Nonnull Resource path,
+                           @Nonnull Resource propertyShape) {
     if (log.isTraceEnabled()) log.trace("(setNodeKind) start");
 
-    boolean hasIri = askTypeProperty(rdfStoreService,
-                                     "nodekind-is-iri.sparql.spel",
-                                     targetClass,
-                                     path);
-
-    boolean hasBlank = askTypeProperty(rdfStoreService,
-                                       "nodekind-is-blank.sparql.spel",
-                                       targetClass,
-                                       path);
-    boolean hasLiteral = askTypeProperty(rdfStoreService,
-                                         "nodekind-is-literal.sparql.spel",
-                                         targetClass,
-                                         path);
+    boolean hasIri = askQuery(rdfStore, "nodekind-is-iri.sparql.spel", getTypeAndPropertyParameters(targetClass, path));
+    boolean hasBlank = askQuery(rdfStore, "nodekind-is-blank.sparql.spel", getTypeAndPropertyParameters(targetClass, path));
+    boolean hasLiteral = askQuery(rdfStore, "nodekind-is-literal.sparql.spel", getTypeAndPropertyParameters(targetClass, path));
```

Comment:
Do we expect the method askQuery to change the paramters Map? Otherwise we can create it once and reuse it...
(if needed we can all Collections.unmodifiableMap())
```suggestion
    Map<String, String> typeAndPropertyParameters = getTypeAndPropertyParameters(targetClass, path);
    boolean hasIri = askQuery(rdfStore, "nodekind-is-iri.sparql.spel", typeAndPropertyParameters);
    boolean hasBlank = askQuery(rdfStore, "nodekind-is-blank.sparql.spel", typeAndPropertyParameters);
    boolean hasLiteral = askQuery(rdfStore, "nodekind-is-literal.sparql.spel", typeAndPropertyParameters);
```